### PR TITLE
Introduce chat.update bindings

### PR DESCRIPTION
**Pull request by lf-**

### CODE REVIEW

File: src/Web/Slack/Chat.hs [hunk 1]
```diff
@@ -106,11 +88,60 @@ mkPostMsgReq channel text =
     }
 
 data PostMsgRsp = PostMsgRsp
-  { postMsgRspTs :: String
+  { postMsgRspTs :: Text
```
Comment (lf-) [inline src/Web/Slack/Chat.hs:92]: This is a breaking change, but it is silly that this was a String to begin with.

File: src/Web/Slack/Chat.hs [hunk 2]
```diff
@@ -59,7 +39,9 @@ $(deriveJSON (jsonOpts "postMsg") ''PostMsg)
 
 data PostMsgReq = PostMsgReq
   { postMsgReqChannel :: Text
-  , postMsgReqText :: Text
+  , postMsgReqText :: Maybe Text
```
Comment (lf-) [inline src/Web/Slack/Chat.hs:43]: This is actually Maybe in the API since you can have a message with blocks and no text.

File: src/Web/Slack.hs [hunk 3]
```diff
@@ -213,6 +218,22 @@ chatPostMessage_ ::
...
+-- <https://api.slack.com/methods/chat.update>
+chatUpdate ::
+  SlackConfig ->
+  Chat.UpdateReq ->
+  IO (Response Chat.UpdateRsp)
+chatUpdate = flip $ \updateReq -> do
+  authR <- mkSlackAuthenticateReq
+  run (chatUpdate_ authR updateReq) . slackConfigManager
+
+chatUpdate_ ::
```
Comment (friedbrice) [inline src/Web/Slack.hs:232]: I feel like I've asked this before about this file, but where are all these "underscored" functions defined?

Comment (lf-) [reply src/Web/Slack.hs:232]: https://github.com/MercuryTechnologies/slack-web/blob/ffeccec09c642b71bf6fee4516e15335c8dfc123/src/Web/Slack.hs#L296-L304

File: src/Web/Slack/Chat.hs (hunk 1 above)
Comment (friedbrice) [reply src/Web/Slack/Chat.hs:92]: will you be bumping the version number?

Comment (lf-) [reply src/Web/Slack/Chat.hs:92]: I am going to write a changelog Soon and release 1.5 (current version is 0.5), but not in this PR.

File: src/Web/Slack/Chat.hs [hunk 4]
```diff
@@ -106,11 +89,60 @@ mkPostMsgReq channel text =
...
   deriving stock (Eq, Generic, Show)
 
 instance NFData PostMsgRsp
 
 $(deriveFromJSON (jsonOpts "postMsgRsp") ''PostMsgRsp)
+
+-- | <https://api.slack.com/methods/chat.update>
+data UpdateReq = UpdateReq
+  { updateReqChannel :: ConversationId
+  , updateReqTs :: Text
```
Comment (friedbrice) [inline src/Web/Slack/Chat.hs:104]: hmm, that's strange. do all messages in a channel have a unique timestamp?

**Review by friedbrice — APPROVED** [review]
The breaking API change is not that bad, people will be able to fix their code easily. I just wonder if it's still a good idea to bump the version number?

File: src/Web/Slack/Chat.hs (hunk 4 above)
Comment (lf-) [reply src/Web/Slack/Chat.hs:104]: yes, lmao. They are the primary key of messages, which is an Interesting design decision on the part of Slack, but I guess, there are many of those.

**Pull request merged**